Geometry_Engine/Query/LineIntersection.cs

```diff
@@ -482,6 +482,76 @@ public static List<Point> LineIntersections(this Polyline curve1, Polyline curve
             return iPts.CullDuplicates(tolerance);
         }
 
+        /***************************************************/
+
+        [Description("Find segments of a line lying inside the region defined by a polyline.")]
+        [Input("line", "A line to check for intersections with a region defined by the input polylines.")]
+        [Input("pLine", "A polyline defining a closed region that the input line potentially intersects.")]
+        [Input("tolerance", "Minimum length required of new intersection lines.")]
+        [Output("intersections", "Segments of the input line that lie inside the region defined by the input polyline.")]
+        public static List<Line> LineIntersections(this Line line, Polyline pLine, double tolerance = Tolerance.Distance)
+        {
+            var intersections = new List<Line>();
+
+            if (line.IsInRange(pLine.Bounds()) == false)
+                return intersections;
+
+            if (pLine.IsContaining(line))
+                return new List<Line> { line };
+
+            List<Point> intersectionPnts = pLine.LineIntersections(line);
+            if (!intersectionPnts.Any())
+                return intersections;
+
+            List<double> pointParameters = intersectionPnts.Select(x => line.ParameterAtPoint(x)).ToList();
+
+            var sortedPointParamPairs = intersectionPnts
+                .Zip(pointParameters, (point, param) => new { point, param })
+                .OrderBy(x => x.param)
+                .Select(x => x).ToList();
+
+            var startPnt = line.PointAtParameter(0);
+            if (pLine.IsContaining(new List<Point> { startPnt }))
+            {
+                var intPnt = sortedPointParamPairs.First().point;
+                if (intPnt.Distance(startPnt) > tolerance)
+                {
+                    intersections.Add(new Line() { Start = startPnt, End = intPnt });
+                }
+            }
+
+            var endPnt = line.PointAtParameter(1);
+            if (pLine.IsContaining(new List<Point> { endPnt }))
+            {
+                var intPnt = sortedPointParamPairs.Last().point;
+                if (intPnt.Distance(endPnt) > tolerance)
+                {
+                    intersections.Add(new Line() { Start = intPnt, End = endPnt });
+                }
+            }
+
+            for (int i = 0; i < (sortedPointParamPairs.Count - 1); i++)
+            {
+                var entry1 = sortedPointParamPairs[i];
+                var entry2 = sortedPointParamPairs[i + 1];
+                var point1 = entry1.point;
+                var point2 = entry2.point;
+
+                if (point1.Distance(point2) < tolerance)
+                    continue;
+
+                double averageParam = (entry1.param + entry2.param) / 2;
+                var midPnt = new List<Point> { line.PointAtParameter(averageParam) };
+
+                if (pLine.IsContaining(midPnt))
```

Comment:
Worth considering the `acceptOnEdge` parameter here as well as tolerance. Should a line that is colinear with an edge and on top of it, within tolerance, be treated as in the region? Probably yes.

Alternatively, worth adding the `acceptOnEdge` as an additional parameter to this method to be passed on here.

Reply:
Good idea! I've added `acceptOnEdge` as a parameter and pass it on to `IsContaining`.